src/Microsoft.SqlTools.ServiceLayer/BatchParser/LexerInput.cs

```diff
@@ -0,0 +1,153 @@
+﻿//------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//------------------------------------------------------------------------------
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.SqlTools.ServiceLayer.BatchParser
+{
+    internal sealed class LexerInput : IDisposable
+    {
+        private readonly string _filename;
```

Comment:
>_filename [](start = 32, length = 9)

can we remove the the '_' from all the field names.  we're not using that coding convention in this code base, so we should try not to introduce new code following a different coding convention when possible.